File: Samples-Xamarin/Fragments/ListView/ListViewReorderFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using Com.Telerik.Widget.List;
using System.Collections;
using Android.Support.V4.App;

namespace Samples
{
	public class ListViewReorderFragment : Fragment, ExampleFragment, ItemReorderBehavior.IItemReorderListener
	{
		private RadListView listView;
	
		public override View OnCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			View rootView = inflater.Inflate(Resource.Layout.fragment_list_view_example, container, false);

			this.listView = (RadListView) rootView.FindViewById(Resource.Id.listView).JavaCast<RadListView>();

			ItemReorderBehavior reorderBehavior = new ItemReorderBehavior();
			reorderBehavior.AddListener (this);
		
			this.listView.AddBehavior(reorderBehavior);

			ArrayList source = new ArrayList();
			source.Add("Item 1");
			source.Add("Item 2");
			source.Add("Item 3");
			source.Add("Item 4");
			source.Add("Item 5");

			this.listView.SetAdapter(new ReorderListViewAdapter(source));

			return rootView;
		}

		public String Title(){
			return "Item reorder";
		}

		public void OnReorderFinished ()
		{

		}

		public void OnReorderItem (int p0, int p1)
		{

		}

		public void OnReorderStarted (int p0)
		{

		}

	}

	class ReorderListViewAdapter : ListViewDataSourceAdapter {

		public ReorderListViewAdapter(IList items) : base(items){

		}

		public override Android.Support.V7.Widget.RecyclerView.ViewHolder OnCreateViewHolder (ViewGroup p0, int p1)
		{
			LayoutInflater inflater = LayoutInflater.From(p0.Context);
			View itemView = inflater.Inflate(Resource.Layout.example_list_view_item_reorder_layout, p0, false);
			ReorderCustomViewHolder customHolder = new ReorderCustomViewHolder(itemView);
			return customHolder;
		}

		public override void OnBindItemViewHolder (ListViewHolder p0, Java.Lang.Object p1)
		{
			ReorderCustomViewHolder customVH = (ReorderCustomViewHolder) p0;
			customVH.txtItemText.Text = p1.ToString();
		}
	}

	class ReorderCustomViewHolder : ListViewHolder {

		public TextView txtItemText;

		public ReorderCustomViewHolder(View itemView) : base(itemView){
			this.txtItemText = (TextView) itemView.FindViewById(Resource.Id.txtItemText);
		}
	}
}

File: Samples-Java/Samples/src/main/java/fragments/listview/ListViewReorderFragment.java
package fragments.listview;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.telerik.android.sdk.R;
import com.telerik.widget.list.ItemReorderBehavior;
import com.telerik.widget.list.ListViewAdapter;
import com.telerik.widget.list.ListViewHolder;
import com.telerik.widget.list.RadListView;

import java.util.ArrayList;
import java.util.List;

import activities.ExampleFragment;

/**
 * A simple {@link Fragment} subclass.
 */
public class ListViewReorderFragment extends Fragment implements ExampleFragment {

    private RadListView listView;

    public ListViewReorderFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_list_view_example, container, false);

        this.listView = (RadListView) rootView.findViewById(R.id.listView);

        ItemReorderBehavior reorderBehavior = new ItemReorderBehavior();
        reorderBehavior.addListener(new ItemReorderBehavior.ItemReorderListener() {
            @Override
            public void onReorderStarted(int i) {
                //Called when the reorder operation is started.
            }

            @Override
            public void onReorderItem(int i, int i2) {
                //Called when the positions of two items are changes following a reorder operation.
            }

            @Override
            public void onReorderFinished() {
                //Called when the user has released the item being reordered.
            }
        });
        this.listView.addBehavior(reorderBehavior);

        ArrayList<String> source = new ArrayList<>();
        source.add("Item 1");
        source.add("Item 2");
        source.add("Item 3");
        source.add("Item 4");
        source.add("Item 5");

        this.listView.setAdapter(new MyListViewAdapter(source));

        return rootView;
    }

    @Override
    public String title() {
        return "Item reorder";
    }

    class MyListViewAdapter extends ListViewAdapter {

        public MyListViewAdapter(List items) {
            super(items);
        }

        @Override
        public ListViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            LayoutInflater inflater = LayoutInflater.from(parent.getContext());
            View itemView = inflater.inflate(R.layout.example_list_view_item_reorder_layout, parent, false);
            MyCustomViewHolder customHolder = new MyCustomViewHolder(itemView);
            return customHolder;
        }

        @Override
        public void onBindViewHolder(ListViewHolder holder, int position) {
            MyCustomViewHolder customVH = (MyCustomViewHolder) holder;
            customVH.txtItemText.setText(this.getItem(position).toString());
        }
    }

    class MyCustomViewHolder extends ListViewHolder {

        public TextView txtItemText;

        public MyCustomViewHolder(View itemView) {
            super(itemView);
            this.txtItemText = (TextView) itemView.findViewById(R.id.txtItemText);
        }
    }
}
